maven-artifact/src/test/java/org/apache/maven/artifact/versioning/ComparableVersionTest.java

```diff
@@ -383,4 +412,15 @@ void testMng7644() {
             checkVersionsEqual("2.0." + x, "2.0.0." + x); // previously ordered, now equals
         }
     }
+
+    @Test
+    public void testMng7714() {
+        String f = "1.0.final-redhat";
```

Comment:
Make these variables ComparableVersions, not strings, and then inline checkVersionsOrder in this method so it's easier to understand in one place what is being tested. `checkVersionsOrder(f, sp1)` is unclear because it does not indicate whether f is expected to be less than, equal to, or greater than sp1.

Reply:
Done.